code-studio/src/js/progress.js

```diff
@@ -181,7 +186,8 @@ progress.renderStageProgress = function (stageData, progressData, clientProgress
   $('.progress_container').replaceWith(mountPoint);
   ReactDOM.render(React.createElement(StageProgress, {
     levels: combinedProgress,
-    currentLevelIndex: currentLevelIndex
+    currentLevelIndex: currentLevelIndex,
+    saveAnswersFirst: puzzlePage !== progress.PUZZLE_PAGE_NONE
```

Comment:
Maybe too late for this feedback, but looking at this all together I wonder if the right pattern would've been to pass a `onDotClicked` callback here.